cmd/kubeadm/app/apis/output/types.go

```diff
@@ -82,3 +84,43 @@ type UpgradePlan struct {
 
 	ConfigVersions []ComponentConfigVersionState
 }
+
+// Certificate represents information about check-expiration for one certificate
+type Certificate struct {
+	// Name of the certificate
+	Name string
+	// ExpirationDate defines certificate expiration date.
+	ExpirationDate metav1.Time
+	// ResidualTime represents the time missing to expiration
+	ResidualTime time.Duration
+	// ExternallyManaged defines if the certificate is externally managed
+	IsExternallyManaged bool
+	// CAName represents the name of the CA that signed the certificate
+	CAName string
+	// IsMissing represents if the certificate is missing
+	IsMissing bool
+}
+
+// CertificateAuthority represents information about check-expiration for one certificate authority
+type CertificateAuthority struct {
+	// Name of the certificate authority
+	Name string
+	// ExpirationDate defines certificate authority expiration date.
+	ExpirationDate metav1.Time
+	// ResidualTime represents the time missing to expiration
+	ResidualTime time.Duration
+	// ExternallyManaged defines if the certificate authority is externally managed
+	IsExternallyManaged bool
+	// IsMissing represents if the certificate authority is missing
+	IsMissing bool
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ExpirationInfo represents information for the output produced by 'kubeadm certs check-expiration'
+type ExpirationInfo struct {
+	metav1.TypeMeta
+
+	Certificates           []Certificate
+	CertificateAuthorities []CertificateAuthority
```

Comment:
```suggestion
	// Certificates holds a list of certificate authorities to show expiration information for.
	Certificates           []Certificate
	
	// CertificateAuthorities holds a list of certificate authorities to show expiration information for.
	CertificateAuthorities []Certificate
```

adding comments above exported fields.